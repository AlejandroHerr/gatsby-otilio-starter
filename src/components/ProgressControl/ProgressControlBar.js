import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styles from './ProgressControl.module.scss';

const ProgressControlBar = ({
  finished, progress, speed, mapPropsToStyle,
}) => (
  <div className={classnames(styles.progress_control_bar, {
        [styles['progress_control_bar--finished']]: finished,
        [styles['progress_control_bar--hide']]: progress <= 0,
      })}
  >
    <div
      className={styles.progress_control_bar__bar}
      style={mapPropsToStyle({ finished, progress, speed })}
    >
      {(progress > 0 && (<div className={styles.progress_control_bar__peg} />)) || null}
    </div>
  </div>
);

ProgressControlBar.defaultProps = {
  finished: false,
};

ProgressControlBar.propTypes = {
  finished: PropTypes.bool,
  progress: PropTypes.number.isRequired,
  speed: PropTypes.number.isRequired,
  mapPropsToStyle: PropTypes.func.isRequired,
};

export default ProgressControlBar;
