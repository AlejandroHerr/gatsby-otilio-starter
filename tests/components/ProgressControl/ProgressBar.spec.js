import React from 'react';
import { shallow } from 'enzyme';

import ProgressControlBar from '../../../src/components/ProgressControl/ProgressControlBar';

const defaultProps = {
  finished: false,
  progress: 0,
  speed: 10,
  mapPropsToStyle: arg => arg,
};

const setup = (props = {}) => {
  const progressControlBar = shallow(<ProgressControlBar {...defaultProps} {...props} />);

  return {
    progressControlBar,
    bar: progressControlBar.find('.progress_control_bar__bar'),
    peg: progressControlBar.find('.progress_control_bar__peg'),
  };
};

describe('ProgressControl', () => {
  describe('ProgressControlBar', () => {
    it('should render ProgressControlBar for progress 0', () => {
      const { progressControlBar, bar, peg } = setup();

      expect(progressControlBar).toHaveLength(1);
      expect(progressControlBar.hasClass('progress_control_bar--hide')).toBeTruthy();
      expect(progressControlBar.hasClass('progress_control_bar--finished')).toBeFalsy();
      expect(bar).toHaveLength(1);
      expect(bar.prop('style')).toEqual(defaultProps.mapPropsToStyle({
        progress: defaultProps.progress,
        speed: defaultProps.speed,
        finished: defaultProps.finished,
      }));
      expect(peg).toHaveLength(0);
      expect(progressControlBar).toMatchSnapshot();
    });
    it('should render ProgressControlBar when in progress', () => {
      const props = {
        finished: false,
        progress: 50,
      };
      const { progressControlBar, bar, peg } = setup(props);

      expect(progressControlBar).toHaveLength(1);
      expect(progressControlBar.hasClass('progress_control_bar--hide')).toBeFalsy();
      expect(progressControlBar.hasClass('progress_control_bar--finished')).toBeFalsy();
      expect(bar).toHaveLength(1);
      expect(bar.prop('style')).toEqual(defaultProps.mapPropsToStyle({
        progress: props.progress,
        speed: defaultProps.speed,
        finished: props.finished,
      }));
      expect(peg).toHaveLength(1);
      expect(progressControlBar).toMatchSnapshot();
    });
    it('should render ProgressControlBar when progress finished', () => {
      const props = {
        finished: true,
        progress: 100,
      };
      const { progressControlBar, bar, peg } = setup(props);

      expect(progressControlBar).toHaveLength(1);
      expect(progressControlBar.hasClass('progress_control_bar--hide')).toBeFalsy();
      expect(progressControlBar.hasClass('progress_control_bar--finished')).toBeTruthy();
      expect(bar).toHaveLength(1);
      expect(bar.prop('style')).toEqual(defaultProps.mapPropsToStyle({
        progress: props.progress,
        speed: defaultProps.speed,
        finished: props.finished,
      }));
      expect(peg).toHaveLength(1);
      expect(progressControlBar).toMatchSnapshot();
    });
  });
});
