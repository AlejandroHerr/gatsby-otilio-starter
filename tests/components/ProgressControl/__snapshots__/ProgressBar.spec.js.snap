// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressControl ProgressControlBar should render ProgressControlBar for progress 0 1`] = `
<div
  className="progress_control_bar progress_control_bar--hide"
>
  <div
    className="progress_control_bar__bar"
    style={
      Object {
        "finished": false,
        "progress": 0,
        "speed": 10,
      }
    }
  />
</div>
`;

exports[`ProgressControl ProgressControlBar should render ProgressControlBar when in progress 1`] = `
<div
  className="progress_control_bar"
>
  <div
    className="progress_control_bar__bar"
    style={
      Object {
        "finished": false,
        "progress": 50,
        "speed": 10,
      }
    }
  >
    <div
      className="progress_control_bar__peg"
    />
  </div>
</div>
`;

exports[`ProgressControl ProgressControlBar should render ProgressControlBar when progress finished 1`] = `
<div
  className="progress_control_bar progress_control_bar--finished"
>
  <div
    className="progress_control_bar__bar"
    style={
      Object {
        "finished": true,
        "progress": 100,
        "speed": 10,
      }
    }
  >
    <div
      className="progress_control_bar__peg"
    />
  </div>
</div>
`;
